src/source.c

```diff
@@ -622,7 +622,7 @@ _dispatch_source_kevent_unregister(dispatch_source_t ds)
 		}
 	} else if (!ds->ds_is_direct_kevent) {
 		ds->ds_dkev = NULL;
-		dispatch_assert(ds->ds_is_installed);
+		dispatch_assert(ds->ds_is_installed == 1);
```

Comment:
I'd rather have a cast to bool. if the bitfield is `int` the value may be -1 and this would be wrong.